src/SWP/Bundle/CoreBundle/Twig/ArticleBodyExtension.php

```diff
@@ -49,9 +49,70 @@ public function getFunctions(): array
     {
         return [
             new TwigFunction('setRendition', [$this, 'setRendition']),
+            new TwigFunction('setInlineAds', [$this, 'setInlineAds']),
         ];
     }
 
+    public function setInlineAds(string $body): array
+    {
+        # split the body into paragraphs
+        $paragraphs = explode("</p>", $body);
+
+        $tmp = [];
+        $splittedBody = [];
+        $parLength = 0;
+        $paragraphChars = $_ENV['ADS_CHARACTERS'];
```

Comment:
You can't do this like that. Publisher (and symfony) have other way for accessing env variables. It should be passed to service via DI.